server/src/main/java/de/uftos/services/ConstraintInstanceService.java

```diff
@@ -177,11 +186,67 @@ public ConstraintInstance update(String signatureId, String id,
    * @param id          the ID of the constraintInstance.
    */
   public void delete(String signatureId, String id) {
-    ConstraintInstance instance = this.getById(signatureId, id);
+    ConstraintInstance instance = this.getInstanceById(signatureId, id);
 
     this.repository.delete(instance);
   }
 
+  private ConstraintInstance getInstanceById(String signatureId, String id) {
+    ConstraintSignature signature = this.signatureRepository.findById(signatureId)
+        .orElseThrow(() -> new ResponseStatusException(HttpStatus.BAD_REQUEST));
+
+    return this.repository.findBySignatureAndId(signature, id)
+        .orElseThrow(() -> new ResponseStatusException(HttpStatus.BAD_REQUEST));
+  }
+
+  private List<ConstraintArgumentDisplayName> processConstraintInstance(
+      ConstraintInstance constraintInstance) {
+    List<ConstraintArgumentDisplayName> displayNames = new ArrayList<>();
+    for (ConstraintArgument constraintArgument : constraintInstance.getArguments()) {
+      ConstraintArgumentDisplayName displayName = getDisplayName(constraintArgument);
+      displayNames.add(displayName);
+    }
+    return displayNames;
+  }
+
+  private List<ConstraintArgumentDisplayName> processConstraintInstances(
+      List<ConstraintInstance> constraintInstances) {
+    List<ConstraintArgumentDisplayName> displayNames = new ArrayList<>();
+    for (ConstraintInstance constraintInstance : constraintInstances) {
+      displayNames.addAll(processConstraintInstance(constraintInstance));
+    }
+    return displayNames;
+  }
+
+  private ConstraintArgumentDisplayName getDisplayName(ConstraintArgument constraintArgument) {
+    String id = constraintArgument.getValue();
+    return switch (constraintArgument.getConstraintParameter().getParameterType()) {
+      case TAG ->
+          new ConstraintArgumentDisplayName(id, tagRepository.findById(id).orElseThrow().getName());
+      case ROOM -> new ConstraintArgumentDisplayName(id,
+          roomRepository.findById(id).orElseThrow().getName());
```

Comment:
We shouldn't have empty `orElseThrow` calls. Either remove them, use `.get()` and accept the runtime error (which I'm not sure if that's fine or not), or throw a bad request.

Reply:
Now throwing a bad request